modules/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/CodeAnalyzer.java

```diff
@@ -212,7 +213,7 @@ private void visitInvocable(BLangInvokableNode invNode) {
                 workerReturns = workerReturns || this.statementReturns;
                 this.resetStatementReturns();
             }
-            if (functionReturns && !workerReturns) {
+            if (invokableReturns && !workerReturns) {
```

Comment:
we have to fix ATLEAST_ONE_WORKER_MUST_RETURN DiagnosticCode similar to  DiagnosticCode.INVOKABLE_MUST_RETURN . Currently, it prints "at least one worker in the **function** must return a result"